### ISSUE TITLE
CAS with ingress annotation fails with panic

### ISSUE THREAD

**Issue by rvennam-lbg** [jetstack/google-cas-issuer#8]
When trying to deploy jenkins using the ingress-shim and cas issuer, with below annotations.  The certificate request get created properly but the google cas issuer pod crashes as soon as the request is created

Ingress object

```yaml
apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      cert-manager.io/issuer: oncp-dev
      cert-manager.io/issuer-group: cas-issuer.jetstack.io
      cert-manager.io/issuer-kind: GoogleCASClusterIssuer
      kubernetes.io/ingress.class: nginx
      meta.helm.sh/release-name: jenkins
      meta.helm.sh/release-namespace: jenkins
      nginx.ingress.kubernetes.io/proxy-body-size: "0"
    creationTimestamp: "2021-01-02T17:25:07Z"
    generation: 1
    labels:
      app.kubernetes.io/component: jenkins-controller
      app.kubernetes.io/instance: jenkins
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: jenkins
      helm.sh/chart: jenkins-3.0.11
    name: jenkins
    namespace: jenkins
    resourceVersion: "169122"
    selfLink: /apis/extensions/v1beta1/namespaces/jenkins/ingresses/jenkins
    uid: ced40d51-0d5a-49fc-8813-24cd95912c8d
  spec:
    rules:
    - host: jenkins-bak-euwe1.mgmt-tst.oncp.dev
      http:
        paths:
        - backend:
            serviceName: jenkins
            servicePort: 8080
    tls:
    - hosts:
      - jenkins-bak-euwe1.mgmt-tst.oncp.dev
      secretName: jenkins-tls
  status:
    loadBalancer:
      ingress:
      - ip: 10.85.128.13
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
  
```
Certificate object

```yaml
apiVersion: v1
items:
- apiVersion: cert-manager.io/v1
  kind: Certificate
  metadata:
    creationTimestamp: "2021-01-02T18:08:13Z"
    generation: 1
    labels:
      app.kubernetes.io/component: jenkins-controller
      app.kubernetes.io/instance: jenkins
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: jenkins
      helm.sh/chart: jenkins-3.0.11
    name: jenkins-tls
    namespace: jenkins
    ownerReferences:
    - apiVersion: extensions/v1beta1
      blockOwnerDeletion: true
      controller: true
      kind: Ingress
      name: jenkins
      uid: ced40d51-0d5a-49fc-8813-24cd95912c8d
    resourceVersion: "169132"
    selfLink: /apis/cert-manager.io/v1/namespaces/jenkins/certificates/jenkins-tls
    uid: 75606156-d113-4aeb-8687-617a4e18e1ed
  spec:
    dnsNames:
    - jenkins-bak-euwe1.mgmt-tst.oncp.dev
    issuerRef:
      group: cas-issuer.jetstack.io
      kind: GoogleCASClusterIssuer
      name: oncp-dev
    secretName: jenkins-tls
  status:
    conditions:
    - lastTransitionTime: "2021-01-02T18:08:13Z"
      message: Issuing certificate as Secret does not exist
      reason: DoesNotExist
      status: "True"
      type: Issuing
    - lastTransitionTime: "2021-01-02T18:08:13Z"
      message: Issuing certificate as Secret does not exist
      reason: DoesNotExist
      status: "False"
      type: Ready
    nextPrivateKeySecretName: jenkins-tls-nbvpb
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

Logs for `google-cas-issuer` are below

```
2021-01-02T17:59:23.612Z	INFO	controller	Starting EventSource	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASClusterIssuer", "controller": "googlecasclusterissuer", "source": "kind source: /, Kind="}
2021-01-02T17:59:23.712Z	INFO	controller	Starting Controller	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASIssuer", "controller": "googlecasissuer"}
2021-01-02T17:59:23.713Z	INFO	controller	Starting workers	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASIssuer", "controller": "googlecasissuer", "worker count": 1}
2021-01-02T17:59:23.713Z	INFO	controller	Starting Controller	{"reconcilerGroup": "cert-manager.io", "reconcilerKind": "CertificateRequest", "controller": "certificaterequest"}
2021-01-02T17:59:23.713Z	INFO	controller	Starting workers	{"reconcilerGroup": "cert-manager.io", "reconcilerKind": "CertificateRequest", "controller": "certificaterequest", "worker count": 1}
2021-01-02T17:59:23.713Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "cert-manager.io", "reconcilerKind": "CertificateRequest", "controller": "certificaterequest", "name": "test-cas-qlndx", "namespace": "default"}
2021-01-02T17:59:23.713Z	INFO	controller	Starting Controller	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASClusterIssuer", "controller": "googlecasclusterissuer"}
2021-01-02T17:59:23.713Z	INFO	controller	Starting workers	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASClusterIssuer", "controller": "googlecasclusterissuer", "worker count": 1}
2021-01-02T17:59:23.713Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "cert-manager.io", "reconcilerKind": "CertificateRequest", "controller": "certificaterequest", "name": "test-stage-le-np6jh", "namespace": "default"}
2021-01-02T17:59:23.716Z	DEBUG	controller	Successfully Reconciled	{"reconcilerGroup": "cas-issuer.jetstack.io", "reconcilerKind": "GoogleCASClusterIssuer", "controller": "googlecasclusterissuer", "name": "oncp-dev", "namespace": ""}
E0102 18:08:13.603404       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 250 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x179c7e0, 0x2686fa0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/runtime/runtime.go:74 +0xa6
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/runtime/runtime.go:48 +0x89
panic(0x179c7e0, 0x2686fa0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/jetstack/google-cas-issuer/pkg/controller/certificaterequest.(*CertificateRequestReconciler).Reconcile(0xc000588c90, 0xc0000511f9, 0x7, 0xc0002e1bc0, 0x11, 0xc00055a010, 0xc000198990, 0xc000198908, 0xc000198900)
	/workspace/pkg/controller/certificaterequest/certificaterequest_controller.go:144 +0xa17
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d4b40, 0x180b4c0, 0xc000159240, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d4b40, 0x203000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001d4b40)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000614480)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000614480, 0x1b9b760, 0xc000654120, 0x1, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000614480, 0x3b9aca00, 0x0, 0x1a7d001, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000614480, 0x3b9aca00, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:170 +0x3fa
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1627f57]

goroutine 250 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x179c7e0, 0x2686fa0)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/jetstack/google-cas-issuer/pkg/controller/certificaterequest.(*CertificateRequestReconciler).Reconcile(0xc000588c90, 0xc0000511f9, 0x7, 0xc0002e1bc0, 0x11, 0xc00055a010, 0xc000198990, 0xc000198908, 0xc000198900)
	/workspace/pkg/controller/certificaterequest/certificaterequest_controller.go:144 +0xa17
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001d4b40, 0x180b4c0, 0xc000159240, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235 +0x2a9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001d4b40, 0x203000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209 +0xb0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001d4b40)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000614480)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000614480, 0x1b9b760, 0xc000654120, 0x1, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000614480, 0x3b9aca00, 0x0, 0x1a7d001, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc000614480, 0x3b9aca00, 0xc0001de240)
	/go/pkg/mod/k8s.io/apimachinery@v0.19.0/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:170 +0x3fa
```

**Comment by rvennam-lbg** [comment]
On further debugging found out the Issue is linked to the missing duration and renewbefore fields which are merged and not yet released and would be part of cert-manger 1.2

PR link https://github.com/jetstack/cert-manager/pull/3465

**Comment by jakexks** [comment]
Hi @rvennam-lbg,
Thanks for the report. This seems like an oversight on my part. I'm currently making a big update to the issuer to move it from a PoC state to something I'd be comfortable releasing and will address any missing fields!
